gcp/src/main/java/org/apache/iceberg/gcp/gcs/GCSInputFile.java

```diff
@@ -64,6 +78,33 @@ public long getLength() {
 
   @Override
   public SeekableInputStream newStream() {
+    if (gcpProperties().isGcsAnalyticsCoreEnabled()) {
+      try {
+        GcsFileInfo fileInfo = getGcsFileInfo();
+        return new GoogleCloudStorageInputStreamWrapper(
+            GoogleCloudStorageInputStream.create(gcsFileSystem(), fileInfo));
+      } catch (IOException e) {
+        LOG.error("Failed to create GCS analytics core  input stream.", e);
+        throw new RuntimeIOException(
+            e, "Failed to create GCS analytics core input stream for: %s", blobId().toGsUtilUri());
+      }
+    }
+
     return new GCSInputStream(storage(), blobId(), blobSize, gcpProperties(), metrics());
   }
+
+  GcsFileInfo getGcsFileInfo() {
+    BlobId blobId = blobId();
+    GcsItemId itemId =
+        GcsItemId.builder()
+            .setBucketName(blobId.getBucket())
+            .setObjectName(blobId.getName())
+            .build();
+    GcsItemInfo itemInfo = GcsItemInfo.builder().setItemId(itemId).setSize(getLength()).build();
```

Comment:
I'm a little concerned about the behavior of the `getLength` call since we may be introducing an extra call just to populate the size.  Is this a required field? Can we provide it only if we know the length as opposed to always fetching?